libsupport/include/katana/OpaqueID.h

```diff
@@ -197,7 +197,12 @@ struct OpaqueIDLinear : public OpaqueIDOrderedWithValue<_IDType, _Value> {
 public:
   using OpaqueIDOrderedWithValue<_IDType, _Value>::OpaqueIDOrderedWithValue;
 
-  using DifferenceType = std::make_signed_t<_Value>;
+  //!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  //Important to note that subtraction between difference types and unsinged integers is only safe as lon as the other integer is 32 bits.
+  //Consider that subtracting two DifferenceTypes could also result in overflow of the difference type.
+  //Need to be careful when subtracting the difference types.
+  using DifferenceType = int64_t; 
+  //!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Comment:
Followed the discussion. Agree with the change. We can stick to 64-bit signed integer as the difference type for any integer type (I'm assuming this is not for float types).

Please fix the comments. Use `///` for doxygen. Drop the excess `!!!..`. Also, add a link to `ptrdiff_t` from the PR in the doxygen code comments.